iabtcf-encoder/src/main/java/com/iabtcf/encoder/Bounds.java

```diff
@@ -0,0 +1,83 @@
+package com.iabtcf.encoder;
+
+/*-
+ * #%L
+ * IAB TCF Java Encoder Library
+ * %%
+ * Copyright (C) 2020 IAB Technology Laboratory, Inc
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import java.util.List;
+
+import com.iabtcf.FieldDefs;
+import com.iabtcf.encoder.exceptions.ValueOverflowException;
+import com.iabtcf.utils.BitSetIntIterable;
+
+class Bounds {
+    public static List<PublisherRestrictionEntry> checkBounds(List<PublisherRestrictionEntry> value) {
+        int max = (1 << FieldDefs.CORE_NUM_PUB_RESTRICTION.getLength()) - 1;
+        if (value.size() > max) {
+            throw new ValueOverflowException(value.size(), max, FieldDefs.CORE_NUM_PUB_RESTRICTION);
+        }
+        return value;
+    }
+
+    /**
+     * This is used to make sure the maximum value is less than a digit of field#getLength number of
+     * bits.
+     */
+    public static BitSetIntIterable.Builder checkBounds(BitSetIntIterable.Builder values, FieldDefs field) {
+        long max = (1L << field.getLength()) - 1;
+        if (values.max() > max) {
```

Comment:
duplicated code. Consider extracitng to separate method

Reply:
fixed